src/mixins/raster-layer-poly-mixin.js

```diff
@@ -253,20 +258,36 @@ export default function rasterLayerPolyMixin(_layer) {
       })
     }
 
-    const colorScaleName = getColorScaleName(layerName)
-    const scales = [
-      {
+    let scales = []
+    let fillColor = {}
+    const useColorScale = state.encoding.color.value === undefined
+    if (useColorScale) {
+      const colorRange = state.encoding.color.range.map(c =>
```

Comment:
We could clean-up to be consistent at some point between using `function() {}` or the fat arrow for anonymous functions